packages/core/crates/core-protocol/src/ticket_aggregation/processor.rs

```diff
@@ -152,29 +152,44 @@ impl AggregationList {
                     .prepare_aggregatable_tickets(&channel_id, epoch, index_start, index_end)
                     .await?
             }
-            AggregationList::TicketList(list) => {
-                if list.is_empty() {
-                    debug!("got empty list of tickets to aggregate");
-                    return Err(ProtocolTicketAggregation("no tickets to aggregate".into()));
-                }
+            AggregationList::TicketList(list) => list,
+        };
 
-                let signer = list[0].signer;
-                let channel_id = list[0].ticket.channel_id;
-
-                if !list.iter().all(|tkt| {
-                    tkt.signer == signer && tkt.ticket.channel_id == channel_id && tkt.status.is_being_aggregated()
-                }) {
-                    error!(
-                        "some tickets to aggregate in the given list do not come from the same channel or are not marked as BeingAggregated"
-                    );
-                    return Err(ProtocolTicketAggregation(
-                        "invalid list of tickets to aggregate given".into(),
-                    ));
-                }
+        if list.is_empty() {
+            debug!("got empty list of tickets to aggregate");
+            return Err(ProtocolTicketAggregation("no tickets to aggregate".into()));
+        }
 
-                list
+        let signer = list[0].signer;
+        let channel_id = list[0].ticket.channel_id;
+
+        let channel_balance = db
+            .read()
+            .await
+            .get_channel(&channel_id)
+            .await?
+            .ok_or(ProtocolTicketAggregation(format!(
+                "channel {channel_id} does not exist"
+            )))?
+            .balance;
+
+        let mut total_amount = Balance::zero(BalanceType::HOPR);
+
+        for tkt in list.iter() {
+            if tkt.signer != signer || tkt.ticket.channel_id != channel_id || !tkt.status.is_being_aggregated() {
```

Comment:
Why are checks on `signer` and `channel_item` necessary? Shouldn't the returned by `prepare_aggregatable_tickets` satisfy those checks?

Reply:
That true for 2/3 cases. The in the 3rd case the caller can pass it is own list of `AcknowledgedTickets` that might not originate from the `prepare_aggregatable_tickets`